backend/routes/user.js

```diff
@@ -444,4 +443,59 @@ user.get("/budget/:userId", authenticate, async (req, res) => {
   }
 });
 
+// adds avoided merchant
+user.post("/avoided-merchant", authenticate, async (req, res) => {
+  const { merchantName } = req.body;
+  const userId = req.user.uid;
+
+  try {
+    const avoidedMerchant = await prisma.avoidedMerchant.create({
+      data: {
+        userId,
+        merchantName: merchantName.trim(),
+      },
+    });
+    res.json(avoidedMerchant);
+  } catch (error) {
+    if (error.code === "P2002") {
```

Comment:
The Prisma error object may not always have a `code` property. This could throw if error is not a Prisma error.
add a safety check: 
```suggestion
    if (error?.code === "P2002") {
```